CONFIGURE: fix AM_GNU_GETTEXT usage

alexey-tikhonov opened the pull request:
Fixed following autoreconf issue:
```
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported
```

alexey-tikhonov committed:
CONFIGURE: fix AM_GNU_GETTEXT usage

Fixed following autoreconf issue:
```
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported
```

elkoniu commented:
Result of `autoreconf -if` without this PR:

```
[koniu@blackbox]<ding-libs><master>$ autoreconf -if
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
configure.ac:58: warning: AM_INTL_SUBDIR is m4_require'd but not m4_defun'd
/usr/share/aclocal/gettext.m4:55: AM_GNU_GETTEXT is expanded from...
configure.ac:58: the top level
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
configure.ac:58: warning: AM_INTL_SUBDIR is m4_require'd but not m4_defun'd
/usr/share/aclocal/gettext.m4:55: AM_GNU_GETTEXT is expanded from...
configure.ac:58: the top level
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
configure.ac:58: warning: AM_INTL_SUBDIR is m4_require'd but not m4_defun'd
aclocal.m4:77: AM_GNU_GETTEXT is expanded from...
configure.ac:58: the top level
autoreconf: configure.ac: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build'.
libtoolize: copying file 'build/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
configure.ac:58: warning: AM_INTL_SUBDIR is m4_require'd but not m4_defun'd
/usr/share/aclocal/gettext.m4:55: AM_GNU_GETTEXT is expanded from...
configure.ac:58: the top level
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
configure.ac:58: warning: AM_INTL_SUBDIR is m4_require'd but not m4_defun'd
/usr/share/aclocal/gettext.m4:55: AM_GNU_GETTEXT is expanded from...
configure.ac:58: the top level
ERROR: Use of AM_GNU_GETTEXT without [external] argument is no longer supported.
configure.ac:58: warning: AM_INTL_SUBDIR is m4_require'd but not m4_defun'd
aclocal.m4:77: AM_GNU_GETTEXT is expanded from...
configure.ac:58: the top level
configure:14665: error: possibly undefined macro: AM_INTL_SUBDIR
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
autoreconf: /usr/bin/autoconf failed with exit status: 1
```

Result of `autoreconf -if` with this PR included:

```
[koniu@blackbox]<ding-libs><master>$ autoreconf -if
autoreconf: configure.ac: AM_GNU_GETTEXT is used, but not AM_GNU_GETTEXT_VERSION
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build'.
libtoolize: copying file 'build/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
configure.ac:8: installing 'build/compile'
configure.ac:57: error: required file 'build/config.rpath' not found
configure.ac:11: installing 'build/missing'
Makefile.am: installing 'build/depcomp'
autoreconf: automake failed with exit status: 1
```

It is better but `autoreconf -if` still fails on Fedora 33.
It looks like for me issue https://github.com/SSSD/ding-libs/issues/8 still exists.

alexey-tikhonov commented:
> It is better but `autoreconf -if` still fails on Fedora 33.
> It looks like for me issue #8 still exists.

But #8 is a different issue. This patch doesn't pretend solving it.

alexey-tikhonov commented:
@sumit-bose , please take a look.

elkoniu commented:
As this PR address the issue it describes I am setting ACK flag.

sumit-bose commented:
Hi,

I agree, the `config.rpath` issue is a different one, please see my comment in #8.

For `AM_GNU_GETTEXT` this patch here looks good.

bye,
Sumit

alexey-tikhonov committed:
CONFIGURE: explicitly require GETTEXT version

This is required by `autopoint` executed by `autoreconf`

Resolves: https://github.com/SSSD/ding-libs/issues/8

alexey-tikhonov commented:
@sumit-bose, @elkoniu, 

I added additional patch per Sumit's suggestion. Please take a look.

elkoniu reviewed (commented):
LGTM and it is compiling now on my Fedora 33

sumit-bose commented:
Hi,

the orignal concerns can be found at https://github.com/SSSD/ding-libs/pull/4#discussion_r161883689. If I understand it correctly the `intl` directory isn't create if `external` is used. The `po` is created by autopoint but since it is not listed in Makefile.am it will not be added to a tar ball (`make dist`). So it will be only present in "developer" environments which imo is acceptable.

Finally there are some additional `m4` files which are added to the tar ball as well. Since `gettext.m4` and `iconv.m4` both contain `AC_REQUIRE([AC_LIB_RPATH])` `config.rpath` must be present and we can either let `autopoint` add it automatically or add `build/config.rpath` explicitly to the git tree. The latter would help to avoid the additional `m4` files in the tar ball. But it looks that typically projects try to avoid to keep such files in git directly. So I prefer the solution from the second patch and set `AM_GNU_GETTEXT_VERSION`.

bye,
Sumit

alexey-tikhonov commented:
Pushed:
 - https://pagure.io/SSSD/ding-libs/c/a12da93b5af0018a2940a4ddde0c1723db866a26
 - https://pagure.io/SSSD/ding-libs/c/868a81b7ad10d21f4895585395540eb04d25cfdf
